### PR TITLE
Add support for CompoundCRS

### PR DESCRIPTION
**Pull request by rhattersley**
## Rationale

Allow Cartopy to make use of EPSG codes that define a compound CRS.

See https://github.com/rhattersley/pyepsg/issues/10.

## Implications

A projection can now be created using an EPSG code for a compound CRS, whereas previously it would have raised an error.

### CODE REVIEW

**Comment by rhattersley** [comment]
Do I need to do something to fix the appveyor problem? Or is #1224 tackling this?

File: lib/cartopy/_epsg.py [hunk 1]
```diff
@@ -41,7 +41,9 @@ class _EPSGProjection(ccrs.Projection):
     def __init__(self, code):
         import pyepsg
         projection = pyepsg.get(code)
-        if not isinstance(projection, pyepsg.ProjectedCRS):
+        if not (isinstance(projection, pyepsg.ProjectedCRS) or
+                (hasattr(pyepsg, 'CompoundCRS') and
```
Comment (pelson) [inline lib/cartopy/_epsg.py]: I'd be extremely tempted to update pyepsg dependency to ``>=0.4``. Relevant locations : https://github.com/SciTools/cartopy/blob/master/requirements/epsg.txt and https://github.com/SciTools/cartopy/blob/master/INSTALL

Comment (pelson) [reply lib/cartopy/_epsg.py]: Let's go for it.

**Comment by rhattersley** [comment]
Should be good to go now. 👍

**Comment by pelson** [comment]
Great, thanks @rhattersley. And welcome back to cartopy contributions 👍 😉

**Comment by rhattersley** [comment]
Nice to be back 😉

**Pull request merged**